addons/core/addon/helpers/time-remaining.js

```diff
@@ -0,0 +1,40 @@
+import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
+import DurationUnitFormat from 'intl-unofficial-duration-unit-format';
+
+export default class extends Helper {
+  // =services
+  @service clockTick;
+  @service intl;
+
+  compute([expiration_time]) {
+    if (!expiration_time) return;
+
+    // Parse the expiration time
+    const expirationTime = new Date(expiration_time);
+
+    // Get the curent time
+    const currentTime = this.clockTick.now;
+
+    // Calculate the difference in milleseconds
+    const difference = expirationTime - currentTime;
+
+    // Calculate the difference in seconds
+    const differenceInSeconds = Math.floor(difference / 1000);
+
+    // If the difference is negative, return 0:0:00 remaining
+    if (differenceInSeconds <= 0)
+      return `0:0:00 ${this.intl.t('resources.session.remaining')}`;
```

Comment:
I am not sure we need a return here. How about if differenceInSeconds <= 0 we do differenceInSeconds = 0 ? then, your implementation below will format correctly `0:00:00` so we avoid manual formatting? thoughts?

Reply:
Perhaps line 23 can be something like: `const differenceInSeconds = (difference > 0) ? Math.floor(difference / 1000) : 0;` and then we avoid line 26 and 27 ?

Reply:
@calcaide awesome that works.  

https://github.com/hashicorp/boundary-ui/assets/24277002/801d98c1-4c8f-42b5-bc40-4a331a0e0cd4